cmake/modules/AddSwift.cmake

```diff
@@ -856,6 +850,31 @@ function(_add_swift_library_single target name)
     set(SWIFTLIB_SINGLE_API_NOTES "${module_name}")
   endif()
 
+  # On Linux and FreeBSD add conformance magic objects that coalesce protocol
+  # conformance sections. This wouldn't be necessary if the link was done by
+  # the swift binary: rdar://problem/19007002
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR
+     "${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+
+    if("${libkind}" STREQUAL "SHARED")
+      set(arch_subdir "${SWIFTLIB_DIR}/${SWIFTLIB_SINGLE_SUBDIR}")
+
+      add_custom_command_target(conformance_object_begin
```

Comment:
I have a proposal, but I am not sure it will work but we can give it a try, the issue is currently this, from what I could gather reading the cmake scripts:
- there's a cycle dependency at source level; this file is being included after AddSwift, cmake can't the objects as such we need to work around it adding the extra add_custom_command to copy the objects.
- there's a dependency inversion, because the objects are effectively built before the add_swift_library is invoked.

The idea is to move the generation of the objects in a separate cmake file, since it is self contained, it could be included in the top level cmake file before AddSwift, at that point cmake should be able to match the dependencies and we can remove the extra copy step. Any thoughts?